### PR TITLE
pip/deps: Bump `pip-tools` -> 6.6.0

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [C] Other... Please describe:

Updates `pip-tools` to latest version 

This may be a breaking change for some as more recent version of `pip-compile` enforce `-` separation of namespace package names (rather than `.`)

OTOH it makes it usable with dependabot which uses (at least by default) a recent version  

## What is the current behavior?

Cant be used with dependabot without a lot of unnecessary changes 

## What is the new behavior?

Hopefully at least - compiled files will be ~same as produced by dependabot so will not keep trying to revert each others changes

## Does this PR introduce a breaking change?

- [X] Yes - of sorts
- [ ] No

`requirements.txt` files compiled with `compile_pip_requirements` may need to be recompiled and/or fail existing test expectations

### CODE REVIEW

**Comment by mattem** [comment]
This is also a breaking change as pip-tools drops support for Python 3.6 in 6.5.0 (I know it's EoL, just a note for the diff) 
Also 6.4.0 requires a minimum of pip 21.2.0 should be noted too.

**Comment by phlax** [comment]
@mattem lmk if i need to add anything, happy to update

testing out the PR i saw a couple of other issues that may cause issues with dependabot 

one is that dependabout runs in the directory of the requirements file so all of the paths are relative to the reqs file not the WORKSPACE

re that issue im wondering if we can add some flag or similar to the rule that allows it run in the directory etc

the other issue im wondering about is the `pip-compile` command in the requirements.txt - its helpful that it shows the correct bazel command, but i think to prevent unnecessary changes with dependabot it needs to just be `pip-compile` 

for ref one of the reqs files i hope to manage with in this way is here https://github.com/envoyproxy/envoy/blob/334974a0081d70b6a258f27cd40a827caf49aabf/tools/base/requirements.txt#L5

im wondering for this case we can make the command customizable in some way

**Comment by f0rmiga** [comment]
@mattem Want to guide @phlax on any specific changes it may need, or could we merge it since Python 3.6 is EOL as you mentioned?

**Comment by phlax** [comment]
@groodt re https://github.com/phlax/rules_python/pull/1#issuecomment-1144195163 i would be happy to add to this PR but as this is a ~breaking change its maybe best to pass this one through first/alone

@f0rmiga any chance of landing this?

its not a total solution to using `rules_python` with dependabot but its a step in the right direction (and i guess to `pip-compiling` more generally)

looks like this needs a main merge, which ill do now...

lmk if there is anything else that needs to be updated

**Comment by groodt** [comment]
@phlax SGTM. Land the merge and this PR and I'll create a fresh new one.

File: python/pip_install/repositories.bzl [hunk 1]
```diff
@@ -29,14 +29,19 @@ _RULE_DEPS = [
     ),
     (
         "pypi__pip_tools",
-        "https://files.pythonhosted.org/packages/6d/16/75d65bdccd48bb59a08e2bf167b01d8532f65604270d0a292f0f16b7b022/pip_tools-5.5.0-py2.py3-none-any.whl",
-        "10841c1e56c234d610d0466447685b9ea4ee4a2c274f858c0ef3c33d9bd0d985",
+        "https://files.pythonhosted.org/packages/ba/42/5a9dfbaac47794c087d115fa275d3248a41614686b84d85cbb0bd146ebe5/pip_tools-6.6.0-py3-none-any.whl",
```
[the remark this answers is not in the document]
Comment (phlax) [reply python/pip_install/repositories.bzl:37]: yep, it was late when i merged, added back now

File: python/pip_install/requirements.bzl [hunk 2]
```diff
@@ -60,9 +60,11 @@ def compile_pip_requirements(
     deps = [
         requirement("click"),
         requirement("colorama"),
+        requirement("pep517"),
         requirement("pip"),
         requirement("pip_tools"),
         requirement("setuptools"),
+        requirement("tomli"),
```
Comment (thundergolfer) [inline python/pip_install/requirements.bzl:67]: How it this one used by pip-tools. I checked the install-requires of pip-tools and it isn't there.

**Review by thundergolfer — APPROVED** [review]

**Comment by phlax** [comment]
im guessing we want a changelog entry somewhere - as this might break some expectations

File: python/pip_install/requirements.bzl (hunk 2 above)
Comment (groodt) [reply python/pip_install/requirements.bzl:67]: `# via pep517`

Comment (thundergolfer) [reply python/pip_install/requirements.bzl:67]: Ah cheers. Didn't expect `pep517` to take any deps.

**Comment by thundergolfer** [comment]
Yes I think this would at least go in the _What's Changed_ section of release notes.

**Comment by groodt** [comment]
Mergeable?

**Pull request merged**